test/vars/MailerSpec.groovy

```diff
@@ -0,0 +1,162 @@
+import com.homeaway.devtools.jenkins.testing.JenkinsPipelineSpecification
+
+class MailerSpec extends JenkinsPipelineSpecification {
+    def groovyScript = null
+
+    def setup() {
+        groovyScript = loadPipelineScriptForTest("vars/mailer.groovy")
+        groovyScript.metaClass.WORKSPACE = '/'
+    }
+
+    def "[mailer.groovy] sendEmailFailure without CHANGE_BRANCH"() {
+        setup:
+        def env = [:]
+        env['ghprbSourceBranch'] = 'PR_1'
+        groovyScript.getBinding().setVariable("env", env)
+        groovyScript.getBinding().setVariable("BUILD_URL", 'https://redhat.com/')
+        when:
+        groovyScript.sendEmailFailure()
+        then:
+        1 * getPipelineMock("emailext")(['subject': 'Build PR_1 failed', body: 'Build PR_1 failed! For more information see https://redhat.com/', 'recipientProviders': [['$class': 'DevelopersRecipientProvider'], ['$class': 'RequesterRecipientProvider']]])
+    }
+
+    def "[mailer.groovy] sendEmailFailure with CHANGE_BRANCH"() {
+        setup:
+        def env = [:]
+        env['CHANGE_BRANCH'] = 'master'
```

Comment:
@Ginxo when we create the 7.39.x (based on master) we have to think about how we can replace 'master' by "7.39.x" since on this branch we also will have PRs etc. - and in shared libraries we can't use another branch, except we create a new shared-library and change the Jenkinsfile on the future 7.39.x

Reply:
@mbiarnes the `CHANGE_BRANCH` variable for the unit tests is just an example, remember they are all tests and its variables are just examples, it could be whatever